src/js/angular-datepicker.js

```diff
@@ -376,7 +376,7 @@
           }
           , unregisterDataSetWatcher = $scope.$watch('dateSet', function dateSetWatcher(newValue) {
 
-            if (newValue) {
+            if (newValue && !isNaN(Date.parse(newValue))) {
```

Comment:
does it fails if you remove isNaN() ?
 i really believe is not necessary but i could mistake actually..